src/Stratis.Bitcoin.Features.WatchOnlyWallet/WatchOnlyWalletManager.cs

```diff
@@ -181,7 +321,8 @@ public WatchOnlyWallet LoadWatchOnlyWallet()
                 Network = this.network,
                 CoinType = this.coinType,
                 CreationTime = this.dateTimeProvider.GetTimeOffset(),
-                WatchedAddresses = new ConcurrentDictionary<string, WatchedAddress>()
+                WatchedAddresses = new ConcurrentDictionary<string, WatchedAddress>(),
+                WatchedTransactions = new ConcurrentDictionary<string, TransactionData>()
```

Comment:
Nice, I think we also need such an index in the wallet itself 😉

Reply:
This is not needed here as it's already done in the WatchOnlyWallet's constructor.